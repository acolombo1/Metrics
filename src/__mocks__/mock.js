const samplepayload = {
  coins: [
    {
      id: 'btc-Bitcoin',
      name: 'Bitcoin',
      symbol: 'BTC',
      rank: 1,
      circulating_supply: 19778208,
      total_supply: 21000000,
      max_supply: 19778208,
      beta_value: 0.7849923,
      first_data_at: '',
      last_updated: '',
      quotes: {
        USD: {
          price: 5691.18449,
          volume_24h: 0,
          volume_24h_change_24h: 0,
          market_cap: 3849999234097,
          market_cap_change_24h: 0,
          percent_change_15m: 0,
          percent_change_30m: 0,
          percent_change_1h: 0,
          percent_change_6h: 0,
          percent_change_12h: 0,
          percent_change_24h: 0,
          percent_change_7d: 0,
          percent_change_30d: 0,
          percent_change_1y: 0,
          ath_price: 68849.29174,
          ath_date: '',
          percent_from_price_ath: 0,
        },
      },
    },
  ],
  globalmktcap: 278000000000,
  coindata: {
    id: '',
    name: '',
    symbol: '',
    rank: 0,
    circulating_supply: 0,
    total_supply: 0,
    max_supply: 0,
    beta_value: 0,
    first_data_at: '',
    last_updated: '',
    quotes: {
      USD: {
        price: 0,
        volume_24h: 0,
        volume_24h_change_24h: 0,
        market_cap: 114567074563,
        market_cap_change_24h: 0,
        percent_change_15m: 0,
        percent_change_30m: 0,
        percent_change_1h: 0,
        percent_change_6h: 0,
        percent_change_12h: 0,
        percent_change_24h: 0,
        percent_change_7d: 0,
        percent_change_30d: 0,
        percent_change_1y: 0,
        ath_price: 0,
        ath_date: '',
        percent_from_price_ath: 0,
      },
    },
  },
};

export const coinsmock = [
  {
    id: 'btc-Bitcoin',
    name: 'Bitcoin',
    symbol: 'BTC',
    rank: 1,
    circulating_supply: 19778208,
    total_supply: 21000000,
    max_supply: 19778208,
    beta_value: 0.7849923,
    first_data_at: '',
    last_updated: '',
    quotes: {
      USD: {
        price: 5691.18449,
        volume_24h: 0,
        volume_24h_change_24h: 0,
        market_cap: 56675474462351,
        market_cap_change_24h: 0,
        percent_change_15m: 0,
        percent_change_30m: 0,
        percent_change_1h: 0,
        percent_change_6h: 0,
        percent_change_12h: 0,
        percent_change_24h: 0,
        percent_change_7d: 0,
        percent_change_30d: 0,
        percent_change_1y: 0,
        ath_price: 68849.29174,
        ath_date: '',
        percent_from_price_ath: 0,
      },
    },
  },
  {
    id: 'eth-Ethereum',
    name: 'Ethereum',
    symbol: 'ETH',
    rank: 2,
    circulating_supply: 1394778208,
    total_supply: 2156000000,
    max_supply: 1394778208,
    beta_value: -0.7849923,
    first_data_at: '',
    last_updated: '',
    quotes: {
      USD: {
        price: 1691.18449,
        volume_24h: 0,
        volume_24h_change_24h: 0,
        market_cap: 156899769769,
        market_cap_change_24h: 0,
        percent_change_15m: 0,
        percent_change_30m: 0,
        percent_change_1h: 0,
        percent_change_6h: 0,
        percent_change_12h: 0,
        percent_change_24h: 0,
        percent_change_7d: 0,
        percent_change_30d: 0,
        percent_change_1y: 0,
        ath_price: 4849.29174,
        ath_date: '',
        percent_from_price_ath: 0,
      },
    },
  },
  {
    id: 'usdt-Tether',
    name: 'Tether',
    symbol: 'USDT',
    rank: 3,
    circulating_supply: 194778208,
    total_supply: 21000023300,
    max_supply: 19778333208,
    beta_value: 0.00149923,
    first_data_at: '',
    last_updated: '',
    quotes: {
      USD: {
        price: 1.18449,
        volume_24h: 0,
        volume_24h_change_24h: 0,
        market_cap: 19778333208,
        market_cap_change_24h: 0,
        percent_change_15m: 0,
        percent_change_30m: 0,
        percent_change_1h: 0,
        percent_change_6h: 0,
        percent_change_12h: 0,
        percent_change_24h: 0,
        percent_change_7d: 0,
        percent_change_30d: 0,
        percent_change_1y: 0,
        ath_price: 1.29174,
        ath_date: '',
        percent_from_price_ath: 0,
      },
    },
  },
  {
    id: 'usdc-USD-Coin',
    name: 'USD Coin',
    symbol: 'USDC',
    rank: 4,
    circulating_supply: 7419778208,
    total_supply: 421000000,
    max_supply: 7419778208,
    beta_value: 1.7849923,
    first_data_at: '',
    last_updated: '',
    quotes: {
      USD: {
        price: 1.00010,
        volume_24h: 0,
        volume_24h_change_24h: 0,
        market_cap: 4938789208,
        market_cap_change_24h: 0,
        percent_change_15m: 0,
        percent_change_30m: 0,
        percent_change_1h: 0,
        percent_change_6h: 0,
        percent_change_12h: 0,
        percent_change_24h: 0,
        percent_change_7d: 0,
        percent_change_30d: 0,
        percent_change_1y: 0,
        ath_price: 1.09174,
        ath_date: '',
        percent_from_price_ath: 0,
      },
    },
  },
  {
    id: 'bnb-Binance-Coin',
    name: 'Binance Coin',
    symbol: 'BNB',
    rank: 5,
    circulating_supply: 619778208,
    total_supply: 210000030,
    max_supply: 197785208,
    beta_value: 9.7849923,
    first_data_at: '',
    last_updated: '',
    quotes: {
      USD: {
        price: 269.18449,
        volume_24h: 0,
        volume_24h_change_24h: 0,
        market_cap: 5197785208,
        market_cap_change_24h: 0,
        percent_change_15m: 0,
        percent_change_30m: 0,
        percent_change_1h: 0,
        percent_change_6h: 0,
        percent_change_12h: 0,
        percent_change_24h: 0,
        percent_change_7d: 0,
        percent_change_30d: 0,
        percent_change_1y: 0,
        ath_price: 549.29174,
        ath_date: '',
        percent_from_price_ath: 0,
      },
    },
  },
];

export default samplepayload;
