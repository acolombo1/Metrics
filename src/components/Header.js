import { useState } from 'react';
import { useDispatch } from 'react-redux';
import './header.css';

function Header() {
  const [filter, updateFilter] = useState({ filter: '' });
  const onFilterChange = (event) => {
    updateFilter({ filter: event.target.value });
  };
  const dispatch = useDispatch();
  return (
    <form className="FilterForm">
      <input value={filter.filter} type="text" id="filter" placeholder="1-10" name="filter" onChange={onFilterChange} />
      <button
        className="FilterButton"
        type="button"
        onClick={() => {
          const filterarr = filter.filter.match(/[0-9]{1,3} *- *[0-9]{1,3}/g);
          if (filterarr) {
            if (filterarr.join('') === filter.filter) {
              const newfilter = filterarr.join().replace(/\s/g, '');
              if (newfilter !== '') {
                const liminf = parseInt(newfilter.substring(0, newfilter.search('-')), 10);
                const limsup = parseInt(newfilter.substring(newfilter.search('-') + 1), 10);
                if (limsup >= liminf) {
                  dispatch({
                    type: 'SET_FILTER',
                    payload: { liminf, limsup },
                  });
                }
              }
            }
          }
        }}
      >
        Filter
      </button>
    </form>
  );
}

export default Header;
