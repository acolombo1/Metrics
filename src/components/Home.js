import PropTypes from 'prop-types';
import './home.css';

function Home(props) {
  const { coins } = props;
  coins.forEach((coin) => console.log(coin));
  return (
    <div className="Home">
      { coins.map((coin) => (
        <span key={coin.id}>
          {coin.id}
          <br />
        </span>
      )) }
    </div>
  );
}

export default Home;

Home.propTypes = {
  coins: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    symbol: PropTypes.string.isRequired,
    rank: PropTypes.number.isRequired,
    circulating_supply: PropTypes.number.isRequired,
    total_supply: PropTypes.number.isRequired,
    max_supply: PropTypes.number.isRequired,
    beta_value: PropTypes.number.isRequired,
    first_data_at: PropTypes.string.isRequired,
    last_updated: PropTypes.string.isRequired,
    quotes: PropTypes.shape({
      USD: PropTypes.shape({
        ath_date: PropTypes.string.isRequired,
        ath_price: PropTypes.number.isRequired,
        market_cap: PropTypes.number.isRequired,
        market_cap_change_24h: PropTypes.number.isRequired,
        percent_change_1h: PropTypes.number.isRequired,
        percent_change_1y: PropTypes.number.isRequired,
        percent_change_6h: PropTypes.number.isRequired,
        percent_change_7d: PropTypes.number.isRequired,
        percent_change_12h: PropTypes.number.isRequired,
        percent_change_15m: PropTypes.number.isRequired,
        percent_change_24h: PropTypes.number.isRequired,
        percent_change_30d: PropTypes.number.isRequired,
        percent_change_30m: PropTypes.number.isRequired,
        percent_from_price_ath: PropTypes.number.isRequired,
        price: PropTypes.number.isRequired,
        volume_24h: PropTypes.number.isRequired,
        volume_24h_change_24h: PropTypes.number.isRequired,
      }),
    }),
  })).isRequired,
};
