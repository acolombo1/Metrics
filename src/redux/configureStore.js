import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import DataReducer from './datareducer';
import FilterReducer from './filterreducer';

const Store = configureStore({
  reducer: {
    coins: DataReducer,
    filter: FilterReducer,
  },
  middleware: [thunk],
});

export default Store;
