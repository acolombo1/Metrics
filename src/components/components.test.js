import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from '../redux/configureStore';
import Header from './Header';
import Home from './Home';
import { coinsmock } from '../__mocks__/mock';

test('Header renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={configureStore}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Home renders correctly for mock data', () => {
  const tree = renderer
    .create(
      <Provider store={configureStore}>
        <MemoryRouter>
          <Home coins={coinsmock} />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
