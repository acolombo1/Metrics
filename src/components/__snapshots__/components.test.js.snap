// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<form
  className="FilterForm"
>
  <input
    id="filter"
    name="filter"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="1-10"
    type="text"
    value=""
  />
  <button
    className="FilterButton"
    id="myBtn"
    onClick={[Function]}
    type="button"
  >
    Filter
  </button>
</form>
`;

exports[`Home renders correctly for mock data 1`] = `
Array [
  <form
    className="FilterForm"
  >
    <input
      id="filter"
      name="filter"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="1-10"
      type="text"
      value=""
    />
    <button
      className="FilterButton"
      id="myBtn"
      onClick={[Function]}
      type="button"
    >
      Filter
    </button>
  </form>,
  <div
    className="Filtered"
  >
    <div
      className="Toprow"
    >
      <div
        className="left"
      >
        <img
          alt="Top 100"
          height={80}
          src="/img/paprika3.png"
          width={80}
        />
      </div>
      <div
        className="right"
      >
        <div>
          TOP 100
           
          <div
            className="smallfont"
          >
            Market cap
          </div>
        </div>
        <div
          className="metric"
        >
          56862.29
           
          Bln
        </div>
      </div>
    </div>
    <div
      className="Secondrow"
    >
      <div>
        Market Caps by COIN
      </div>
    </div>
    <div
      className="Remainingrows"
    >
      <a
        className="square"
        href="/btc-Bitcoin"
        onClick={[Function]}
      >
        <div>
          <img
            alt="BTC"
            height={50}
            src="/img/BTC.png"
            width={50}
          />
          <br />
           
          <br />
          <div
            className="toright"
          >
            BTC
             
            <span
              className="smallfont toright"
            >
              Market cap
            </span>
          </div>
          <div
            className="toright"
          >
            56675.47 Bln
          </div>
        </div>
      </a>
      <a
        className="square"
        href="/eth-Ethereum"
        onClick={[Function]}
      >
        <div>
          <img
            alt="ETH"
            height={50}
            src="/img/ETH.png"
            width={50}
          />
          <br />
           
          <br />
          <div
            className="toright"
          >
            ETH
             
            <span
              className="smallfont toright"
            >
              Market cap
            </span>
          </div>
          <div
            className="toright"
          >
            156.9 Bln
          </div>
        </div>
      </a>
      <a
        className="square"
        href="/usdt-Tether"
        onClick={[Function]}
      >
        <div>
          <img
            alt="USDT"
            height={50}
            src="/img/USDT.png"
            width={50}
          />
          <br />
           
          <br />
          <div
            className="toright"
          >
            USDT
             
            <span
              className="smallfont toright"
            >
              Market cap
            </span>
          </div>
          <div
            className="toright"
          >
            19.78 Bln
          </div>
        </div>
      </a>
      <a
        className="square"
        href="/usdc-USD-Coin"
        onClick={[Function]}
      >
        <div>
          <img
            alt="USDC"
            height={50}
            src="/img/USDC.png"
            width={50}
          />
          <br />
           
          <br />
          <div
            className="toright"
          >
            USDC
             
            <span
              className="smallfont toright"
            >
              Market cap
            </span>
          </div>
          <div
            className="toright"
          >
            4.94 Bln
          </div>
        </div>
      </a>
      <a
        className="square"
        href="/bnb-Binance-Coin"
        onClick={[Function]}
      >
        <div>
          <img
            alt="BNB"
            height={50}
            src="/img/BNB.png"
            width={50}
          />
          <br />
           
          <br />
          <div
            className="toright"
          >
            BNB
             
            <span
              className="smallfont toright"
            >
              Market cap
            </span>
          </div>
          <div
            className="toright"
          >
            5.2 Bln
          </div>
        </div>
      </a>
    </div>
    <div
      className="Footer"
    >
      <div>
         
      </div>
    </div>
  </div>,
]
`;
